### PR TITLE
Add docs examples

### PR DESCRIPTION
**Pull request by moritzwilksch**
- Adding 5 examples in docstrings in `frame.py` and `expr.py`
- Fix typo in description of `.arr` namespace

### CODE REVIEW

File: py-polars/polars/internals/frame.py [hunk 1]
```diff
@@ -3528,8 +3551,34 @@ def melt(
         value_vars
             Values to use as identifier variables.
 
-        Returns
-        -------
+        Examples
+        --------
```
Comment (ritchie46) [inline py-polars/polars/internals/frame.py:3555]: Could you add a line between the `Examples` and the start of the code snippets?

**Comment by ritchie46** [comment]
Thanks a lot! I have one small remark.

File: py-polars/polars/internals/frame.py (hunk 1 above)
Comment (moritzwilksch) [reply py-polars/polars/internals/frame.py:3555]: Sure. Does this apply to all `Examples` headings?

Comment (ritchie46) [reply py-polars/polars/internals/frame.py:3555]: Yeah.. It is a recommendation of numpy doc. We don't check it in CI, so here is the manual nit. :P

Comment (moritzwilksch) [reply py-polars/polars/internals/frame.py:3556]: Ok, added the lines :D

**Pull request merged**